plugins/record-hook/src/RecordHook.ts

```diff
@@ -116,99 +98,37 @@ export const BulkRecordHook = async (
       const data = await event.cache.get<Flatfile.RecordsWithLinks>('data')
       const modifiedRecords: Flatfile.RecordsWithLinks = records.filter(
         (record: Flatfile.RecordWithLinks) => {
-          const originalRecord: Flatfile.RecordWithLinks = data.find(
-            (original: Flatfile.RecordWithLinks) => original.id === record.id
-          )!
-          return hasChange(record, originalRecord)
+          const originalRecord: Flatfile.RecordWithLinks | undefined =
+            data.find(
+              (original: Flatfile.RecordWithLinks) => original.id === record.id
+            )
+          cleanRecord(originalRecord) // Remove fields that should not be compared
+          const hasChanges = !deepEqual(record, originalRecord)
+          if (debug) {
+            logInfo(
+              '@flatfile/plugin-record-hook',
+              `Record ${record.id} ${
+                hasChanges ? 'has' : 'does not have'
+              } changes`
+            )
+          }
+          return hasChanges
         }
       )
 
       if (!modifiedRecords || modifiedRecords.length === 0) {
-        if (options.debug) {
-          console.log('No records modified')
-        }
-        await completeCommit()
-        return
+        logInfo('@flatfile/plugin-record-hook', 'No records modified')
+        return await completeCommit(event, debug)
```

Comment:
Wondering if we want to adjust this since the function actually only returns when trackChanges is true -